pkg/query-service/app/querier/querier.go

```diff
@@ -469,7 +470,115 @@ func (q *querier) runClickHouseQueries(ctx context.Context, params *v3.QueryRang
 }
 
+type logsListTsRange struct {
+	Start int64
+	End   int64
+}
+
+const HOUR_NANO = int64(3600000000000)
+
+func getLogsListTsRanges(start, end int64) []logsListTsRange {
+	startNano := utils.GetEpochNanoSecs(start)
+	endNano := utils.GetEpochNanoSecs(end)
+	result := []logsListTsRange{}
+
+	if endNano-startNano > HOUR_NANO {
+		bucket := HOUR_NANO
+		tStartNano := endNano - bucket
+
+		complete := false
+		for {
+			result = append(result, logsListTsRange{Start: tStartNano, End: endNano})
+			if complete {
+				break
+			}
+
+			bucket = bucket * 2
+			endNano = tStartNano
+			tStartNano = tStartNano - bucket
+
+			// break condition
+			if tStartNano <= startNano {
+				complete = true
+				tStartNano = startNano
+			}
+		}
+	}
+	return result
+}
+
+func (q *querier) runLogsListQuery(ctx context.Context, params *v3.QueryRangeParamsV3, keys map[string]v3.AttributeKey, tsRanges []logsListTsRange) ([]*v3.Result, map[string]error, error) {
+	res := make([]*v3.Result, 0)
+	qName := ""
+	pageSize := uint64(0)
+
+	// se we are considering only one query
+	for name, v := range params.CompositeQuery.BuilderQueries {
+		qName = name
+		pageSize = v.PageSize
+	}
+	data := []*v3.Row{}
+
+	for _, v := range tsRanges {
+		params.Start = v.Start
+		params.End = v.End
+
+		params.CompositeQuery.BuilderQueries[qName].PageSize = pageSize - uint64(len(data))
+		queries, err := q.builder.PrepareQueries(params, keys)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// this will to run only once
+		for name, query := range queries {
+			rowList, err := q.reader.GetListResultV3(ctx, query)
+			if err != nil {
+				errs := []error{err}
+				errQuriesByName := map[string]error{
+					name: err,
+				}
+				return nil, errQuriesByName, fmt.Errorf("encountered multiple errors: %s", multierr.Combine(errs...))
+			}
+			data = append(data, rowList...)
+		}
+
+		// append a filter to the params
+		if len(data) > 0 {
+			params.CompositeQuery.BuilderQueries[qName].Filters.Items = append(params.CompositeQuery.BuilderQueries[qName].Filters.Items, v3.FilterItem{
+				Key: v3.AttributeKey{
+					Key:      "id",
+					IsColumn: true,
+					DataType: "string",
+				},
+				Operator: v3.FilterOperatorLessThan,
+				Value:    data[len(data)-1].Data["id"],
+			})
+		}
+
+		if uint64(len(data)) >= pageSize {
+			break
+		}
+	}
+	res = append(res, &v3.Result{
+		QueryName: qName,
+		List:      data,
+	})
+	return res, nil, nil
+}
+
 func (q *querier) runBuilderListQueries(ctx context.Context, params *v3.QueryRangeParamsV3, keys map[string]v3.AttributeKey) ([]*v3.Result, map[string]error, error) {
+	// List query has support for only one query.
+	if q.UseLogsNewSchema && params.CompositeQuery != nil {
+		for _, v := range params.CompositeQuery.BuilderQueries {
+			// only allow of logs queries with timestamp ordering desc
+			if v.DataSource == v3.DataSourceLogs && len(v.OrderBy) == 1 && v.OrderBy[0].ColumnName == "timestamp" && v.OrderBy[0].Order == "desc" {
```

Comment:
What happens in other cases?

Reply:
In other cases, the entire time range will be used. this logic will not be applied.

Reply:
When the order is "asc" why can't we apply the same logic?

Reply:
let me solve that separately in a different issue, as this will be the majority of users use-case.

https://github.com/SigNoz/signoz/issues/5943